sdk/dapp-kit/src/hooks/networkConfig.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { SuiClientOptions } from '@mysten/sui.js/client';
+
+import { useSuiClientContext } from './useSuiClient.js';
+
+export function createNetworkConfigs<T extends Record<string, SuiClientOptions>>(
+	networkConfigs: T,
+): { networkConfigs: T; useNetworkConfig: () => T[keyof T] } {
+	return {
+		networkConfigs,
+		useNetworkConfig: () => {
```

Comment:
I might just not fully understand, but what is the purpose of having this function return a hook versus just having `useNetworkConfig` readily available? I'm guessing it's so we get the strong typing with the `as T[keyof T]` cast?

Reply:
Yes, it's basically just for type-safety.  I personally like this better than making someone pass type arguments